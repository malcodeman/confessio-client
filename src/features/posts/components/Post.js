import React from "react";
import styled from "styled-components";
import { distanceInWordsToNow } from "date-fns";
import { Link } from "react-router-dom";

const Article = styled.article`
  background-color: #fff;
  display: flex;
  flex-direction: column;
  margin-bottom: 40px;
`;

const Header = styled.header`
  padding: 16px 10px;
  border-top: 1px solid #e6e6e6;
  border-right: 1px solid #e6e6e6;
  border-left: 1px solid #e6e6e6;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

const ProfilePhoto = styled.img`
  height: 30px;
  width: 30px;
  border-radius: 50%;
  object-fit: cover;
  margin-right: 10px;
`;

const NameFirstLetter = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 30px;
  width: 30px;
  border-radius: 50%;
  object-fit: cover;
  margin-right: 10px;
  color: #fff;
  font-size: 0.8rem;
  background-color: #007aff;
`;

const StyledLink = styled(Link)`
  display: flex;
  align-items: center;
`;

const Text = styled.span`
  font-size: 0.8rem;
  color: rgba(0, 0, 0, 0.8);
`;

const Photo = styled.img`
  max-width: 100%;
  object-fit: cover;
`;

const Time = styled.time`
  font-size: 0.6rem;
  color: rgba(0, 0, 0, 0.6);
  text-transform: uppercase;
`;

const Footer = styled.footer`
  padding: 16px 10px;
  border-right: 1px solid #e6e6e6;
  border-bottom: 1px solid #e6e6e6;
  border-left: 1px solid #e6e6e6;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

const Post = props => {
  return (
    <Article>
      <Header>
        <StyledLink to={`/${props.username}`}>
          {props.profilePhotoURL ? (
            <ProfilePhoto src={props.profilePhotoURL} />
          ) : (
            <NameFirstLetter>{props.nameFirstLetter}</NameFirstLetter>
          )}
          <Text>{props.username}</Text>
        </StyledLink>
      </Header>
      <Photo src={props.photoURL} />
      <Footer>
        <Time>{distanceInWordsToNow(props.createdAt)} ago</Time>
      </Footer>
    </Article>
  );
};

export default Post;
