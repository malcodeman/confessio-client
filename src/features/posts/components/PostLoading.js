import React from "react";
import styled from "styled-components";

const Article = styled.div`
  margin-bottom: 40px;
  padding: 10px;
  border-radius: ${props => props.theme.borderRadius};
  border: 1px solid ${props => props.theme.borderColor};
  background-color: ${props => props.theme.backgroundSecondary};
`;

const Text = styled.div`
  height: 0.8rem;
  width: 25%;
  margin-bottom: 10px;
  background-color: ${props => props.theme.backgroundPrimary};
`;

const Time = styled.div`
  height: 0.6rem;
  width: 10%;
  background-color: ${props => props.theme.backgroundPrimary};
`;

const PostLoading = () => {
  return (
    <Article>
      <Text />
      <Time />
    </Article>
  );
};

export default PostLoading;
