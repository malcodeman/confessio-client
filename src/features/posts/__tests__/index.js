import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import "jest-styled-components";

import Post from "../components/Post";
import PostLoading from "../components/PostLoading";

const testPost = {
  username: "test_username",
  profilePhotoURL:
    "https://instagram.fsjj1-1.fna.fbcdn.net/vp/f08e4b6b37fe1f96a606668534c898b9/5BBD97E2/t51.2885-15/e35/35132935_188808391837094_8704942949919621120_n.jpg",
  comments: [
    { id: 1, body: "comment 1", user: { username: "admin" } },
    { id: 2, body: "comment 2", user: { username: "test" } }
  ],
  liked: true,
  bookmarked: true,
  nameFirstLetter: "t",
  photoURL:
    "https://instagram.fprg2-1.fna.fbcdn.net/vp/28fd04be5b090942578954037710c8ab/5BCFD69E/t51.2885-15/e35/35273985_403106216853758_9170321326664056832_n.jpg",
  likesCount: 33,
  createdAt: "2018-08-04 06:20:40"
};

test("Post matches snapshot", () => {
  const component = renderer.create(
    <MemoryRouter>
      <Post
        createdAt={testPost.createdAt}
        likesCount={testPost.likesCount}
        bookmarked={testPost.bookmarked}
        photoURL={testPost.photoURL}
        liked={testPost.liked}
        username={testPost.username}
        profilePhotoURL={testPost.profilePhotoURL}
        comments={testPost.comments}
      />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("PostLoading matches snapshot", () => {
  const component = renderer.create(<PostLoading />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
