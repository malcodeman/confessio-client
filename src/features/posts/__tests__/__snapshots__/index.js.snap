// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches snapshot 1`] = `
.c7 {
  padding: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  height: 20px;
  width: 20px;
  cursor: pointer;
}

.c9 {
  padding-bottom: 10px;
}

.c10 {
  color: rgba(0,0,0,1);
  font-size: 0.8rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 10px;
}

.c12 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.8rem;
  padding: 4px 0;
}

.c14 {
  color: #262626;
  margin-right: 10px;
}

.c15 {
  color: rgba(0,0,0,0.8);
}

.c17 {
  border-top: 1px solid rgba(0,0,0,0.1);
  padding: 10px 0;
}

.c18 {
  color: #262626;
  height: 36px;
  font-size: 0.8rem;
  padding: 0;
  outline: 0;
  border: 0;
  width: 100%;
}

.c0 {
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 40px;
}

.c1 {
  padding: 16px 10px;
  border-top: 1px solid #e6e6e6;
  border-right: 1px solid #e6e6e6;
  border-left: 1px solid #e6e6e6;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
}

.c3 {
  height: 30px;
  width: 30px;
  border-radius: 50%;
  object-fit: cover;
  margin-right: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 0.8rem;
  color: rgba(0,0,0,0.8);
}

.c5 {
  max-width: 100%;
  object-fit: cover;
}

.c16 {
  font-size: 0.6rem;
  color: rgba(0,0,0,0.6);
  text-transform: uppercase;
  padding-bottom: 10px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 10px;
  border-right: 1px solid #e6e6e6;
  border-bottom: 1px solid #e6e6e6;
  border-left: 1px solid #e6e6e6;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
}

<article
  className="c0"
>
  <header
    className="c1"
  >
    <a
      className="c2"
      href="/test_username"
      onClick={[Function]}
    >
      <img
        className="c3"
        src="https://instagram.fsjj1-1.fna.fbcdn.net/vp/f08e4b6b37fe1f96a606668534c898b9/5BBD97E2/t51.2885-15/e35/35132935_188808391837094_8704942949919621120_n.jpg"
      />
      <span
        className="c4"
      >
        test_username
      </span>
    </a>
  </header>
  <img
    className="c5"
    src="https://instagram.fprg2-1.fna.fbcdn.net/vp/28fd04be5b090942578954037710c8ab/5BCFD69E/t51.2885-15/e35/35273985_403106216853758_9170321326664056832_n.jpg"
  />
  <footer
    className="c6"
  >
    <section
      className="c7"
    >
      <div>
        <img
          className="c8"
          data-cy="unlike-btn"
          onClick={[Function]}
          src="heart_liked.svg"
        />
        <img
          className="c8"
          src="comment.svg"
        />
      </div>
      <img
        className="c8"
        data-cy="unbookmark-btn"
        onClick={[Function]}
        src="bookmarked.svg"
      />
    </section>
    <section
      className="c9"
    >
      <span
        className="c10"
      >
        33
         likes
      </span>
    </section>
    <section
      className="c11"
    >
      <ul
        className="c12"
      >
        <li
          className="c13"
        >
          <a
            className="c14"
            href="/admin"
            onClick={[Function]}
          >
            admin
          </a>
          <span
            className="c15"
          >
            comment 1
          </span>
        </li>
        <li
          className="c13"
        >
          <a
            className="c14"
            href="/test"
            onClick={[Function]}
          >
            test
          </a>
          <span
            className="c15"
          >
            comment 2
          </span>
        </li>
      </ul>
    </section>
    <time
      className="c16"
    >
      about 8 years
       ago
    </time>
    <section
      className="c17"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="c18"
          name="body"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Add a comment..."
          type="text"
          value=""
        />
      </form>
    </section>
  </footer>
</article>
`;

exports[`PostLoading matches snapshot 1`] = `
.c0 {
  border-radius: 3px;
  border: 1px solid #e6e6e6;
  background-color: #fff;
  margin-bottom: 40px;
  padding: 10px;
}

.c1 {
  height: 0.8rem;
  background-color: rgba(0,0,0,0.1);
  width: 25%;
  margin-bottom: 10px;
}

.c2 {
  height: 0.6rem;
  background-color: rgba(0,0,0,0.1);
  width: 10%;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
</div>
`;
